### PR TITLE
Clean up flydowns

### PR DESCRIPTION
**Pull request by BeksOmega**
### Description

I did some reformatting which means this will probably be easiest to review commit-wise. Here is a slightly more detailed position of each commit:

Commit 1: Reformats the field_parameter_flyout.js file so that it is more readable.
Commit 2: Removes the unused opt_codename & this.eventParam code. These were supposed to be used for internationalization, but they are not. Instead getEventParam creates a map (well parallel arrays, but same thing) of translated names -> untranslated names and uses that to find the event param.
Commit 3: Deletes the procedure flydown because it is unused.
Commit 4: Reformats the field_flydown.js file so that it is more readable.
Commit 5: Adds a new event flydown. This handles component event parameters.
Commit 6: Makes displayLocation optional (since it was already being used that way).
Commit 7: Fixes the position of event parameter flydowns when they are arranged horizontally.

Currently on master the flydown is always displayed to the right.
Horiz
Vertical

Which doesn't match procedures.
ProcedureHorizontal
ProcedureVertical

#### I18n
1. Created the following blocks:
TestBlocks

2. Switched to Spanish:
Spanish

3. Observed that the blocks matched the behavior in production.

#### Arrange Horiz/Vert Option
Observed that the arrange option comes before the collapse option:
BeforeCollapsed

Observed that if a return procedure has variables the inline inputs option is removed:
NoInline

Observed that if a return procedures /does not have/ variables the inline inputs option is available:
WithInline

### Additional Info

While technically this is blocking me from adding dropdown blocks in flydowns, that is not a super high priority.

### CODE REVIEW

File: appinventor/blocklyeditor/src/field_procedure_flydown.js
Comment (fturbak) [inline appinventor/blocklyeditor/src/field_procedure_flydown.js]: That appears to be correct!

File: appinventor/blocklyeditor/src/field_parameter_flydown.js
Comment (BeksOmega) [reply appinventor/blocklyeditor/src/field_parameter_flydown.js:20]: Yes :D I am following the [google's javascript styleguide](https://google.github.io/styleguide/jsguide.html) which is also the guide the App Inventor uses. In particular I am making these files conform to the [column limit](https://google.github.io/styleguide/jsguide.html#formatting-column-limit) and [linewrapping](https://google.github.io/styleguide/jsguide.html#formatting-line-wrapping) rules which greatly aid readability.

There are other things in App Inventor's javascript code which could be changed, eg naming convensions, comment formating, more thorough jsdoc. But I really feel like the line wrapping gives you the best return on investment.

**Comment by BeksOmega** [comment]
Thank you so much for the review! :D

**Comment by BeksOmega** [comment]
> This commit undoes the changes made by Evan in this commit to handle bugs in dealing with correctly translating event parameters when switching languages:
> 
> [8b2a23f#diff-20e8c6e84696fbc2852bd7a3bea8ccd8](https://github.com/mit-cml/appinventor-sources/commit/8b2a23f7a3bd55b803421a81c3787fe22667a688#diff-20e8c6e84696fbc2852bd7a3bea8ccd8)
> 
> Do you have some reason to think that these changes are not necessary?

A couple actually :D Firstly on this branch I am not able to replicate #703 so we know that isn't an issue. Secondly the way these fields have been i18n-ed has changed a lot over the years. There have been attempts to fix that issue (#708) at least 4 times I think? And the complexity of the system has continued to grow. It is no longer necessary to have mutators in this specific context, because the updateMutation() function sets the block's eventparam property when setValue() is called.

While it is not necessary to add a mutator here, it may be helpful for clarity of the system. I will move that logic into the EventParameterFlydown, since it is only used by event prameters.

>You are correct that this is not used anywhere. My plan at the time (circa 2014) was that hovering over the procedure name in a procedure declaration would create a flydown containing the caller of the procedure, so you wouldn't have to pull that out of the procedure drawer. I still think that's a good idea, but it's confusing for this file to exist until that behavior is actually implemented.

I'll file an issue to track that :D

> I don't understand why event flydowns need to be distinguished from other parameter flydowns. Right now FieldEventFlydown is just a thin veneer around FieldParameterFlydown. Is there some future use in your project that justifies the distinction?

Yessir. We want to add helper blocks to flydowns, but only event parameter flydowns.

File: appinventor/blocklyeditor/src/field_flydown.js
Comment (BeksOmega) [reply appinventor/blocklyeditor/src/field_flydown.js]: [Returning early](https://medium.com/better-programming/are-early-returns-any-good-eed4b4d03866) is a just a programming style which I tend to like. I do think it generally improves clarity, and I think having an entire method wrapped in an if statement leads to worse code if the complexity of this function ever grows. But if you don't like it I can revert.

File: appinventor/blocklyeditor/src/field_flydown.js
Comment (BeksOmega) [reply appinventor/blocklyeditor/src/field_flydown.js]: I'm not sure that there's a convention in this situation. But then again I don't necessarily think of it as approving commits. I think of it more like more like you're approving the whole PR, but you're just looking at the changes in the PR in a way that makes it easier to review them (in this case commit-wise).

File: appinventor/blocklyeditor/src/field_event_flydown.js
Comment (BeksOmega) [reply appinventor/blocklyeditor/src/field_event_flydown.js:24]: Those are part of the Closure type system. {TypeName=} means the param is optional. {!TypeName} means the param must be non-null. The full extent of the Closure type system is quite deep so don't feel the need to read the whole thing, but [here](https://github.com/google/closure-compiler/wiki/Types-in-the-Closure-Type-System) is where you can find the documentation.

File: appinventor/blocklyeditor/src/field_flydown.js
Comment (BeksOmega) [reply appinventor/blocklyeditor/src/field_flydown.js:42]: It's not necessary in this case because [the logic](https://github.com/mit-cml/appinventor-sources/blob/c796aca85faa57f69ba214767a20636f9c0a3645/appinventor/blocklyeditor/src/field_flydown.js#L176) uses and else for the default.
```
  if (this.displayLocation === Blockly.FieldFlydown.DISPLAY_BELOW) {
    y = y + borderBBox.height * flydown.workspace_.scale;
  } else { // if (this.displayLocation === Blockly.FieldFlydown.DISPLAY_RIGHT) {
    x = x + borderBBox.width * flydown.workspace_.scale;
  }
```

We could make it explicit if you want though.

File: appinventor/blocklyeditor/src/field_event_flydown.js
Comment (BeksOmega) [reply appinventor/blocklyeditor/src/field_event_flydown.js:24]: Also note that this App Inventor isn't very strict about type annotations, but I figured I could fix these ones up while I was in the area.

File: appinventor/blocklyeditor/src/field_flydown.js
Comment (fturbak) [reply appinventor/blocklyeditor/src/field_flydown.js]: Fair enough. A key benefit I see is using less indentation. The notion of getting unusual cases out of the way early is also compelling.

File: appinventor/blocklyeditor/src/field_event_flydown.js
Comment (fturbak) [reply appinventor/blocklyeditor/src/field_event_flydown.js:24]: Thanks for the explanation and link. I didn't know about this before, but now I do, thanks to you!

File: appinventor/blocklyeditor/src/field_flydown.js
Comment (fturbak) [reply appinventor/blocklyeditor/src/field_flydown.js:42]: Understanding the default behavior via the logic you highlighted is subtle and nonobvious. Please instead use the conditional expression I recommended.

File: appinventor/blocklyeditor/src/field_flydown.js
Comment (fturbak) [reply appinventor/blocklyeditor/src/field_flydown.js]: I want to resolve this comment thread (and others), but can't figure out how to do that. Please see my email on this.

File: appinventor/blocklyeditor/src/field_flydown.js
Comment (BeksOmega) [reply appinventor/blocklyeditor/src/field_flydown.js]: [Done =)](https://github.com/mit-cml/appinventor-sources/pull/2215/commits/2ad8a205d6ab74ac7327de118baa9f78ecf897a6)

**Comment by BeksOmega** [comment]
@ewpatton I think this is ready for merging!

**Comment by ewpatton** [comment]
@BeksOmega Sure. We just want to wait for the dust to settle from tonight's release before we start merging new changes.

**Review by ewpatton — APPROVED** [review]

**Pull request merged**
